xapian-core/cluster/cluster.cc

```diff
@@ -296,10 +296,10 @@ Point::Point(const TermListGroup &tlg, const Document &document_)
 	string term = *it;
 	double termfreq = tlg.get_termfreq(term);
 
-	/** If term is not a stemmed term or the term indexes only one
-	 *  one document, do not compute TF-IDF scores for them
-	 */
-	if (wdf < 1 || term[0] != 'Z' || termfreq <= 1)
+	/// If term frequency of the term is lesser than or equal
+	/// to 1 or wdf is lesser than 1, we do not consider this
+	/// term for document vector calculations
+	if (wdf < 1 || termfreq <= 1)
 	    continue;
```

Comment:
I thought we were going to leave the hardcoded term name check for now?

A couple of points about the comment here:

The first is that `/**` and `///` mark it as a documentation comment for doxygen to handle, but this isn't a documentation comment (in that sense) - it should just use `/*` or `//`.

The second is that the current comment pretty much just says what the code says, but more verbosely (and confusingly with the condition reordered compared to the code).  Generally you should assume the reader has a working knowledge of the programming language so they'll know that `wdf < 1 || termfreq <= 1` means if "wdf is less than 1 or termfreq is less than or equal to 1", and `continue;` skips the current term - by spelling such things out you don't help the reader understand the code, but you do risk having the comment and the code disagree and it being unclear which is right! (If there's a more esoteric language feature in use, or something common is used in an unusual way that can be worth noting).

If you're going to have a comment here, focus on **why**, because that's harder to understand from just the code.  We skip terms with `termfreq <= 1` because such terms only occur in one document (or not at all, if that's even possible here) and so aren't useful as a dimension to cluster in (or at least that's my understanding of the reason).  And `wdf < 1` because ... well, I'm not entirely sure why (which is why it would be good to explain).

The comment before ("the term indexes only one one document") at least hinted at the "why" in a way which "term frequency of the term is lesser than or equal to 1 or wdf is lesser than 1" doesn't, but a more explicit explanation would be better.